sr_data_visualization/src/sr_data_visualization/sr_data_visualizer_gui.py

```diff
@@ -54,14 +57,18 @@ def __init__(self, context):
         self.init_ui()
 
     def _detect_hand_id_and_joints(self):
-        self.joint_prefix = None
+        self.joint_prefix = []
         self.hand_joints = None
 
         try:
             joint_states_msg = rospy.wait_for_message("/joint_states", JointState, timeout=1)
-            self.joint_prefix = ([prefix for prefix in joint_states_msg.name if 'h_' in prefix][0]).split("_")[0] + "_"
-            joints = [joint for joint in joint_states_msg.name if self.joint_prefix in joint]
-            self.hand_joints = {self.joint_prefix[:-1]: joints}
+            for prefix in ['rh_', 'lh_']:
+                for joint in joint_states_msg.name:
+                    if prefix in joint:
+                        self.joint_prefix.append(prefix)
```

Comment:
if i understand correctly we want self.joint_prefix to always be 1 element with prefix. so wouldn't it be better to just have joint_prefix = prefix? the code will be nicer instead of always grabbing the 0 element of the joint_prefix list. (unless we can have more than one?)

Reply:
I changed the behaviour of that variable so that it contains all prefixes that are detected. If there was only one side, it would be rh_ or lh_, but if there were both sides running the list would contain rh_ and lh_. Afterwards the list is used to create the options of the combobox.